pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -106,23 +109,54 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-        if instance.Spec.KubeVirtImagePullPolicy != "" {
-           reqLogger := log.WithValues("imagePullPolicy", instance.Spec.KubeVirtImagePullPolicy)
-           reqLogger.Info("HCO CR contains KubeVirt Image Pull Policy")
-           KubeVirtImagePullPolicy = instance.Spec.KubeVirtImagePullPolicy
-        }
+	if instance.Spec.KubeVirtImagePullPolicy != "" {
+		reqLogger := log.WithValues("imagePullPolicy", instance.Spec.KubeVirtImagePullPolicy)
+		reqLogger.Info("HCO CR contains KubeVirt Image Pull Policy")
+		_, err = verifyPullPolicyType(instance.Spec.KubeVirtImagePullPolicy)
+		if err != nil {
+			reqLogger := log.WithValues("HCO CR imagePullPolicy", instance.Spec.KubeVirtImagePullPolicy)
+			reqLogger.Info("HCO CR KubeVirt Pull Policy is invalid.")
+			instance.Status.KubeVirt = err.Error()
+			KubeVirtCRError = true
+		} else {
+			KubeVirtImagePullPolicy = instance.Spec.KubeVirtImagePullPolicy
+			instance.Status.KubeVirt = "Pull Policy type is valid."
+			KubeVirtCRError = false
+		}
+	}
+
+	if instance.Spec.CDIImagePullPolicy != "" {
+		reqLogger := log.WithValues("imagePullPolicy", instance.Spec.CDIImagePullPolicy)
+		reqLogger.Info("HCO CR contains CDI Image Pull Policy")
+		_, err = verifyPullPolicyType(instance.Spec.CDIImagePullPolicy)
+		if err != nil {
+			reqLogger := log.WithValues("HCO CR imagePullPolicy", instance.Spec.CDIImagePullPolicy)
+			reqLogger.Info("HCO CR CDI Pull Policy is invalid.")
+			instance.Status.CDI = err.Error()
+			CDICRError = true
+		} else {
+			CDIImagePullPolicy = instance.Spec.CDIImagePullPolicy
+			instance.Status.CDI = "Pull Policy type is valid."
+			CDICRError = false
+		}
+	}
 
-        if instance.Spec.CDIImagePullPolicy != "" {
-           reqLogger := log.WithValues("imagePullPolicy", instance.Spec.CDIImagePullPolicy)
-           reqLogger.Info("HCO CR contains CDI Image Pull Policy")
-           CDIImagePullPolicy = instance.Spec.CDIImagePullPolicy
-        }
+	r.updateStatus(instance)
+
+	if KubeVirtCRError || CDICRError {
+		return reconcile.Result{}, err
+
+	}
 
 	// Define a new KubeVirt object
+	reqLogger.Info("Creating a new KubeVirt object")
 	virtCR := newKubeVirtForCR(instance)
 
 	// Set HyperConverged instance as the owner and controller
 	if err := controllerutil.SetControllerReference(instance, virtCR, r.scheme); err != nil {
+		reqLogger.Info(err.Error())
```

Comment:
> 
> 
> why don't you reqLogger.Error?

Thought about it and decided that there is no much difference in this case.

Reply:
I think that if we are logging something that is an error ("failed to set controller reference"), it should not go to info ("XY was created"). Maybe it does not change anything with the current logging library, but if we switch to something that actually writes ERROR or INFO in the beginning of the log, it should go to ERROR. Also, it will make sure that it will be always logged even with higher level of logging. It will be obvious which event caused the reconcile to repeat etc.